doc/adr/0015-current-API-authentication-strategy.md

```diff
@@ -0,0 +1,116 @@
+# 0015. A description of the current API authentication strategy
+
+## Status
+
+Accepted
+
+## Problem
+
+We have no documentation around how authentication works in Weave Gitops. This will aim to capture the current state of
+the system.
+
+The 'decision' section has been skipped as the decisions have been made and reverse-engineering the rationale for them is unlikely to work. The 'context' section covers implementation details and the 'consequences' aims to enumerate major risks with the system as well as how to work with it safely.
+
+This supersedes the sections of [ADR-0005 Wego Core Auth Strategy](./0005-wego-core-auth-strategy.md) not to do with authentication to GitHub/GitLab.
+
+## Context
+
+**Note:** throughout this I refer to "auth methods" and "methods" which, for the sake of this document, mean the general process of authenticating who a request is coming from NOT the specific object method which may be called.
```

Comment:
if we are aiming for authentication, we might want to call it authN to make explicit that `auth` does not include authorization (authZ)